src/zenon.cpp

```diff
@@ -310,69 +347,143 @@ void createEventPoolAndEvents(ze_context_handle_t& context,
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[35], { &kernel_ts_event[34] }, 1, 260);                              //<-res4d_branch2a
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[36], { &kernel_ts_event[35] }, 1, 603);                              //<-res4d_branch2b
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[37], { &kernel_ts_event[36] }, 1, 267);                              //<-res4d_branch2c
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[38], { &kernel_ts_event[37] }, 1, 260);                              //<-res4e_branch2a
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[39], { &kernel_ts_event[38] }, 1, 603);                              //<-res4e_branch2b
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[40], { &kernel_ts_event[39] }, 1, 267);                              //<-res4e_branch2c
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[41], { &kernel_ts_event[40] }, 1, 260);                              //<-res4f_branch2a
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[42], { &kernel_ts_event[41] }, 1, 603);                              //<-res4f_branch2b
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[43], { &kernel_ts_event[42] }, 1, 267);                              //<-res4f_branch2c
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[44], { &kernel_ts_event[43] }, 1, 543);                              //<-res5a_branch1
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[45], { &kernel_ts_event[43] }, 1, 175);                              //<-res5a_branch2a
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[46], { &kernel_ts_event[45] }, 1, 603);                              //<-res5a_branch2b
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[47], { &kernel_ts_event[46] }, 1, 265);                              //<-res5a_branch2c
+            submit_kernel_to_cmd_list(cmp_bound_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[48], { &kernel_ts_event[44], &kernel_ts_event[47] }, 2, 287);        //<-res5b_branch2a
```

Comment:
counter should have better API. you shuold pass ns or us on the API and scaling into expected number of cycles/time of execution should be done in submit_kernel_to_cmd_list. something like
if (cmp_bound_kernel)
counter *= 129
else
counter *= 532
otherwise it would be hard to use it in the future. No one knows your magic formula